src/sync/sync_session.cpp

```diff
@@ -427,13 +427,19 @@ void SyncSession::handle_progress_update(uint64_t downloaded, uint64_t downloada
         m_current_downloadable = downloadable;
         m_current_downloaded = downloaded;
         m_current_uploaded = uploaded;
+        std::vector<uint64_t> to_erase;
         for (auto& pair : m_notifiers) {
             bool should_delete = false;
-            invocations.emplace_back(create_notifier_invocation(pair.second, should_delete));
+            auto package = pair.second;
```

Comment:
This creates a copy of the package. Is that intentional? Is the local variable necessary for some reason?